nasa/earthdata-search
# EDSC-3597: Convert AuthRequiredContainer and ButtonDropdown to functional components.

# Overview

### What is the feature?

Convert AuthRequiredContainer and ButtonDropdown class components to use React hooks and remove code pertaining to CollectionContainer.

### What is the Solution?

Use useState and useEffect to substitute UNSAFE component will mount and will receive props methods. Remove references to CollectionContainer and CollectionDownloadDisplay since they are outmoded by OrderStatus code.

### What areas of the application does this impact?

Any sections that require login to access: admin, contact info, preferences, downloads, subscriptions. ButtonDropdown is not currently used anywhere in Earthdata-Search so it doesn't impact anything and should maybe be removed? CollectionContainer does not render on any pages so its removal does not impact anything.

# Testing

### Reproduction steps

To test AuthRequiredContainer:
1. Go to http://localhost:8080/search and attempt to access any of the following pages
1. a. http://localhost:8080/admin
1. b. http://localhost:8080/contact-info
1. c.  http://localhost:8080/downloads
1. d. http://localhost:8080/preferences
1. e. http://localhost:8080/subscriptions
2. The page should take you through authentication before your desired page appears.

There's no real way to test ButtonDropdown since it's not referenced anywhere in Earthdata-Search.

To test that CollectionContainer has been removed safely:
1. Add some granules to a project and go to their download page
2. Verify that the page you arrive at (http://localhost:8080/downloads/<your-project-id>) still looks the same as it does in prod (https://search.earthdata.nasa.gov/downloads/<different-project-id>)

### Attachments

Add granules to a project
<img width="1707" alt="image" src="https://github.com/nasa/earthdata-search/assets/45649183/dc4abb2a-12d1-4dae-85f3-0cfb06b7536e">
Go to your project
<img width="1714" alt="image" src="https://github.com/nasa/earthdata-search/assets/45649183/94106ca5-6875-4929-9a86-f310f38d6037">
Download data
<img width="1500" alt="image" src="https://github.com/nasa/earthdata-search/assets/45649183/d3593bc8-eec6-4a95-9aea-699b8259f8a2">
Download page should look like this
<img width="1727" alt="image" src="https://github.com/nasa/earthdata-search/assets/45649183/ac0470d6-fc77-40a2-8af1-dc69966127a2">

# Checklist
